packages/dev-server-esbuild/test/ts.test.ts

```diff
@@ -243,4 +243,31 @@ class Bar {
       server.stop();
     }
   });
+
+
+  it('reads tsconfig.json file', async () => {
+    const { server, host } = await createTestServer({
+      rootDir: path.join(__dirname, 'fixture'),
+      plugins: [
+        {
+          name: 'test',
+        },
+        esbuildPlugin({ ts: true, tsconfig: path.join(__dirname, 'fixture', 'tsconfig.json') }),
+      ],
+    });
+
+    try {
+      const response = await fetch(`${host}/a/b/foo.ts`);
+      const text = await response.text();
+
+      expect(response.status).to.equal(200);
+      expect(response.headers.get('content-type')).to.equal(
+        'application/javascript; charset=utf-8',
+      );
+
+      expectIncludes(text, '// a comment');
```

Comment:
Because of `"removeComments": false` in the tsconfig.json I expect that comment remain in the output. The plugin is reading the file and passing the contents to ESBuild, but ESBuild doesn't seem to acknowledge. 

I'm probably missing something silly.

Reply:
Fixed https://github.com/modernweb-dev/web/pull/1922#issuecomment-1087835781